src/main/kotlin/org/wycliffeassociates/otter/jvm/recorder/device/AudioRecorder.kt

```diff
@@ -0,0 +1,81 @@
+package org.wycliffeassociates.otter.jvm.recorder.device
+
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import org.wycliffeassociates.otter.common.device.IAudioRecorder
+import javax.sound.sampled.AudioFormat
+import javax.sound.sampled.AudioSystem
+
+class AudioRecorder : IAudioRecorder{
+
+    private val monitor = Object()
+
+    @Volatile
+    private var stop = false
+
+    companion object {
+        val SAMPLE_RATE = 44100F // Hz
+        val SAMPLE_SIZE = 16 // bits
+        val CHANNELS = 1
+        val SIGNED = true
+        val BIG_ENDIAN = false
+        val FORMAT = AudioFormat(
+            SAMPLE_RATE,
+            SAMPLE_SIZE,
+            CHANNELS,
+            SIGNED,
+            BIG_ENDIAN
+        )
+        val BUFFER_SIZE = 1024
+    }
+
+    private var line = AudioSystem.getTargetDataLine(FORMAT)
+    private val audioByteObservable = PublishSubject.create<ByteArray>()
+    private val recordingStream = Observable.fromCallable {
+        val byteArray = ByteArray(BUFFER_SIZE)
+        var totalRead = 0
+            while (true) {
+                if(line.isOpen || line.available() > 0) {
+                    totalRead += line.read(byteArray, 0, byteArray.size)
+                    audioByteObservable.onNext(byteArray)
+                } else {
+                    synchronized(monitor) {
```

Comment:
I'm pretty confused what's going on with this monitor and why it is synchronized. Are you just ensuring that only one thread can be executing the while loop?

Reply:
@KJoslyn Only one thread should be hitting this block of code, so synchronization isn't really needed to guard the _reads_.  This code is launched by the object's instantiation as it creates the `recordingStream` field.  Sure, it's passed off to a thread pool because of the `subscribeOn` below, but there's still just one thread hitting this block of code per instance of `AudioRecorder`.

This monitor is more so this thread can go to sleep if the line isn't open, and will be woken up if someone hits start/stop/pause.  It's similar to avoiding the `while(!joy) {}` mentioned in the link above.

Reply:
@aunger Thanks for the explanation- this makes complete sense now!